homeassistant/components/onewire/onewirehub.py

```diff
@@ -15,21 +17,38 @@ class OneWireHub:
     def __init__(self, hass: HomeAssistantType):
         """Initialize."""
         self.hass = hass
+        self.owproxy: protocol._Proxy = None
 
-    async def can_connect(self, host, port) -> bool:
-        """Test if we can authenticate with the host."""
+    async def connect(self, host: str, port: int) -> None:
+        """Connect to the owserver host."""
         try:
-            await self.hass.async_add_executor_job(protocol.proxy, host, port)
-        except (protocol.Error, protocol.ConnError) as exc:
-            _LOGGER.error(
-                "Cannot connect to owserver on %s:%d, got: %s", host, port, exc
+            self.owproxy = await self.hass.async_add_executor_job(
+                protocol.proxy, host, port
             )
-            return False
-        return True
+        except protocol.ConnError as exc:
+            raise CannotConnect from exc
 
-    async def is_valid_mount_dir(self, mount_dir) -> bool:
+    async def check_mount_dir(self, mount_dir: str) -> None:
         """Test that the mount_dir is a valid path."""
         if not await self.hass.async_add_executor_job(os.path.isdir, mount_dir):
-            _LOGGER.error("Cannot find directory %s", mount_dir)
-            return False
-        return True
+            raise InvalidPath
+
+    async def initialize(self, config_entry: ConfigEntry) -> None:
+        """Initialize a config entry."""
+        if config_entry.data[CONF_TYPE] == CONF_TYPE_SYSBUS:
+            await self.check_mount_dir(config_entry.data[CONF_MOUNT_DIR])
+        elif config_entry.data[CONF_TYPE] == CONF_TYPE_OWSERVER:
+            host = config_entry.data[CONF_HOST]
+            port = config_entry.data[CONF_PORT]
+            try:
+                await self.connect(host, port)
+            except CannotConnect as exc:
+                raise PlatformNotReady() from exc
```

Comment:
raise `ConfigEntryNotReady` instead. 

It feels a bit weird that this is handled inside this file, and not inside `async_setup_entry`.